staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -84,6 +94,71 @@ func NewListWatchFromClient(c Getter, resource string, namespace string, fieldSe
 	return &ListWatch{ListFunc: listFunc, WatchFunc: watchFunc}
 }
 
+// TODO: really think if it's worth it
+func NewListWatchFromMethods(listerWatcher interface{}, fieldSelector fields.Selector) *ListWatch {
```

Comment:
I wonder how people feel about this function?

In one way it's really useful and simplifying for its users but hard to maintain.

It also move semantic check from compile time to runtime, but Go just doesn't have templates :/